screens/ProductRowforOffer.js

```diff
@@ -0,0 +1,279 @@
+                          updatedProducts: this.state.uniqueselectedProducts,
+                          isPinned: this.props.products['pinned'].filter(pa => pa.id == data.id)[0] !== undefined ? true : false
+                        })
+                      }} style={{ marginRight: 50 }}>{data.name}</Text>
+                  }
+
+                  <TouchableOpacity onPress={() => this.handlepinToggle(data.id)} style={[{ position: 'absolute', right: 24 }]}>
+                    {
+                      <AntDesignIcon
+                        name={'pushpin'}
+                        size={22}
+                        color={
+                          data.pinned
+                            ? mainThemeColor
+                            : '#ccc'}
+                        style={{ transform: [{ rotateY: '180deg' }], marginTop: 18 }} />
+                    }
+                  </TouchableOpacity>
+                </View>
+              )
+            })}
+          </View>
+        }
+      </View>
+    );
+  };
+
+  handleReArrange = (data) => {
+    var from = data.from;
+    var to = data.to;
+
+    var previousProductLabelId = this.state.data[from].id;
+    var nextProductLabelId = this.state.data[to].id;
+
+    var changedPosition = {};
+    changedPosition.index = to - 1;
+
+    if (changedPosition.index < 0) {
+      return false;
+    } else if (changedPosition.index === 0) {
+      changedPosition.previousProductLabelId = null;
+    } else {
+      changedPosition.previousProductLabelId = previousProductLabelId
+    }
+
+    if (changedPosition.index === this.state.data.length - 3) {
+      changedPosition.nextProductLabelId = null;
+    } else if (changedPosition.index > this.state.data.length - 3) {
+      return false;
+    } else {
+      changedPosition.nextProductLabelId = nextProductLabelId
+    }
+
+    previousProductLabelId !== undefined &&
+      dispatchFetchRequest(
+        api.productLabel.sortPrdList(previousProductLabelId),
+        {
+          method: 'POST',
+          withCredentials: true,
+          credentials: 'include',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify(changedPosition)
+        },
+        response => {
+          successMessage('Success')
+          this.props.getLables()
+          this.props.getProducts()
+        }
+      ).then()
+  }
+
+  render() {
+    const {
+      labels = [],
+      navigation
+    } = this.props
+    const { t } = this.context
+    var getlabels = !labels ? navigation.state.params.labels : labels
+    var labelsArr = [{ label: 'pinned', id: 'pinned' }, ...getlabels, { label: 'ungrouped', id: 'ungrouped' }]
+
+    return (
+      <DismissKeyboard>
+        <View style={[styles.fullWidthScreen, styles.nomgrBottom]}>
+          <ScreenHeader backNavigation={true}
+            title={t('productListTitle')}
+            parentFullScreen={true}
+          />
+          <DraggableFlatList
```

Comment:
don't use draggable flat list. use normal flat list.

Reply:
ok, lll change, but in my side not crash happen by using this.